File: Bella/Bella.Services/Database/OrderItem.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Bella.Services.Database
{
    public class OrderItem
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [Range(1, int.MaxValue)]
        public int Quantity { get; set; }
        
        [Required]
        public decimal UnitPrice { get; set; } // Price at time of purchase
        
        [Required]
        public decimal TotalPrice { get; set; } // UnitPrice * Quantity
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        // Foreign keys
        [Required]
        public int OrderId { get; set; }
        
        [Required]
        public int ProductId { get; set; }
        
        // Navigation properties
        public virtual Order Order { get; set; } = null!;
        public virtual Product Product { get; set; } = null!;
    }
}

File: Bella/UI/bella_desktop/lib/model/order_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_item.g.dart';

@JsonSerializable()
class OrderItem {
  final int id;
  final int quantity;
  final double unitPrice;
  final double totalPrice;
  final DateTime createdAt;
  final int orderId;
  final int productId;
  final String productName;
  final String? productPicture;

  OrderItem({
    this.id = 0,
    this.quantity = 0,
    this.unitPrice = 0.0,
    this.totalPrice = 0.0,
    required this.createdAt,
    this.orderId = 0,
    this.productId = 0,
    this.productName = '',
    this.productPicture,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => _$OrderItemFromJson(json);
  Map<String, dynamic> toJson() => _$OrderItemToJson(this);
}

